src/services/NumberService.ts

```diff
@@ -21,29 +21,34 @@ export class TestNumberService {
      * @param attempt - the current number of attempts for generating a Test Number.
      * @param awsError - AWS error to be passed when the request fails otherwise null.
      */
-    public createTestNumber(attempts: number, awsError: AWSError|null): Promise<TestNumber> {
+    public createNumber(attempts: number, awsError: AWSError | null, numberType: NUMBER_TYPE.TRAILER_ID | NUMBER_TYPE.TEST_NUMBER = NUMBER_TYPE.TEST_NUMBER): Promise<TestNumber | TrailerId> {
```

Comment:
a separate `createTrailerId` function is cleaner and implements single responsibility principle. Internally it can call a common function shared by `createTestNumber`